lib/hackerone/client.rb

```diff
@@ -52,6 +55,44 @@ def program
         @program || HackerOne::Client.program
       end
 
+      def programs
+        response = self.class.hackerone_api_connection.get do |req|
+          req.url "me/programs"
+        end
+
+        data = JSON.parse(response.body, :symbolize_names => true)[:data]
+        if data.nil?
+          raise RuntimeError, "Expected data attribute in response: #{response.body}"
+        end
+
+        data.map do |program|
+          Program.new(program)
+        end
+      end
+
+      def program_as_object
+        program_handle_we_want = program
+        programs.find do |program|
+          program.handle == program_handle_we_want
+        end
+      end
+
+      def reporters
+        raise ArgumentError, "Program cannot be nil" unless program
+        response = self.class.hackerone_api_connection.get do |req|
+          req.url "programs/#{program_as_object.id}/reporters"
+        end
+
+        data = JSON.parse(response.body, :symbolize_names => true)[:data]
```

Comment:
How will this behave if the response is a 4xx or 5xx? I think we should reuse `parse_response` here and just have it return a hash (instead of a report).